python map and filter builtin func additions (#603)
Codecademy/docs: content/python/concepts/built-in-functions/terms/filter/filter.md

---
Title: 'filter()'

> Description: 'A built-in function that applies a function to each item in an iterable and returns the values that are true.'

Reviewer: ```suggestion
Description: 'Applies a function to each item in an iterable and returns the values that are true.'
```

For term entires, let's make sure to begin the meta-`Description` with a verb.

Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
---

> 'A built-in function that applies a function to each item in an interable and returns the values that are true. The output of `filter()` is a filter object, in the examples below `list()` is applied to the filter expressions to return a new list.'

Reviewer: ```suggestion
The `filter()` function returns a filter object that contains values filtered from an iterable where a function returns `True` for a given item's value.
```

## Syntax

> ```py

Reviewer: ```suggestion
```pseudo
```

filter(func_name, iterable)
```

> 

Reviewer: Let's add 2-3 sentences here (or a list) that describes what each function argument is and does. 😄

> ## Example 1

Reviewer: ```suggestion
## Example
```

For Docs entries, we should only include one Codebyte snippet towards the end of the entry.

Use `filter()` to return all the odd values in the following list:

> ```codebyte/python

Reviewer: ```suggestion
```py
```

For Docs entries, we should only include one Codebyte snippet towards the end of the entry.

nums = [33, 99, 63, 29, 25, 96, 61, 25, 22, 89, 90, 90]

> print(list(filter(lambda x: x % 2 != 0, nums)))

Reviewer: Could this be re-written to not use the `lambda` keyword? Maybe separate out the function that checks for odd numbers?

```py
nums = [33, 99, 63, 29, 25, 96, 61, 25, 22, 89, 90, 90]

def check_odd(n):
  if n % 2 != 0:
    return True
  else:
    return False

print(list(filter(check_odd, nums)))
```

Alternatively, using `== 1`, instead:

```py
def check_odd(n):
  if n % 2 ==1:
    return True
  else:
    return False
```

> # Output: [33, 99, 63, 29, 25, 61, 25, 89]```

Reviewer: Let's put that trailing ``` on the next line by itself.

> ## Example 2

Reviewer: ```suggestion
## Codebyte Example
```

> Use `filter()` to return only the strings that are all lowercase:

Reviewer: ```suggestion
`filter()` can be used to return only the strings that are all lowercase:
```

Author: I imagine the standards have evolved since the start of docs, but early on examples were usually framed as an imperative like this - https://www.codecademy.com/resources/docs/javascript/strings/replace

Has there been a move away from this approach?

Reviewer: Yeah I think if I recall correctly from prior discussions, we want to move away from any/all usage of "we" and "you" pronouns (even ones implied with imperatives) to make the Docs entries more manual-like.

Reviewer: That is to say, the entry you cited should also be edited at some point in the future with the imperatives removed and replaced.

```codebyte/python
some_strings = ['Learn','How', 'to', 'Code', 'Online']

list(filter(lambda x: x.islower(), some_strings))

> # Output: ['to']```

Reviewer: Same thing here, can we move ``` to the next line?

Reviewer: ```suggestion
```
```

---
Title: 'filter()'
Description: 'Returns a filter object that applies a function to each item in an iterable and returns the values that are True.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
---

> The `filter()` function returns a filter object that contains values filtered from an iterable where a function returns `True` for a given item's value.

Reviewer 2: ```suggestion
The `filter()` function returns a filter object that contains values from an iterable. If a function returns `True` for a given item's value, that value will be included in the returned object.
```

> 'The `filter()` function returns a filter object that contains values filtered from an iterable where a function returns `True` for a given item's value.'

Reviewer 3: ```suggestion
The `filter()` function returns a filter object that contains values filtered from an iterable where a function returns `True` for a given item's value.
```

---
Title: 'filter()'
Description: 'Returns a filter object that applies a function to each item in an iterable and returns the values that are True.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
---

The `filter()` function returns a filter object that contains values filtered from an iterable where a function returns `True` for a given item's value.

## Syntax

> ```psuedo

Reviewer: ```suggestion
```pseudo
```

filter(func_name, iterable)
```

The `func_name` is the function to be applied to the iterable and can be a lambda function or the name of any defined function. The `iterable` contains the items, such as a list, the function will act on.

> ## Example 1

Reviewer: ```suggestion
## Example
```

> Use `filter()` to return all the odd values in the following list:

Reviewer: ```suggestion
The following example uses `filter()` to return all the odd values in the `nums` list:
```

For Docs entries, they should have a voice/tone that is declarative and not imperative.

```py
nums = [33, 99, 63, 29, 25, 96, 61, 25, 22, 89, 90, 90]

def odds_test(n):
    n = True if n % 2 == 1 else False
    return n

> print(list(filter(odds_test, nums)))
> 

Reviewer: ```suggestion
print(list(filter(odds_test, nums)))
```

# Output: [33, 99, 63, 29, 25, 61, 25, 89]
```

> ## Example 2

Reviewer: ```suggestion
## Codebyte Example
```

In most cases, and especially for term entries, sections that feature Codebyte snippets should have a header named "Codebyte Example". Newer term entries seem to have been following this layout.

> `filter()` can be used to return only the strings that are all lowercase:

Reviewer: Like this here is a good example of using declarative phrasing to preview a new concept, code snippet, etc. 👍

---
Title: 'filter()'
Description: 'Returns a filter object that applies a function to each item in an iterable and returns the values that are True.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
---

The `filter()` function returns a filter object that contains values from an iterable. If a function returns `True` for a given item's value, that value will be included in the returned object.

## Syntax

```pseudo
filter(func_name, iterable)
```

> The `func_name` is the function to be applied to the iterable and can be a lambda function or the name of any defined function. The `iterable` contains the items, such as a list, the function will act on.

Reviewer 2: ```suggestion
- The `func_name` is the function to be applied to the iterable and can be a lambda function or the name of any defined function.
- The `iterable` contains the items, such as a list, the function will act on.
```

Reviewer 2: What do we think about changing this to bullet points?
